internal/testdata/options/test_editions.proto

```diff
@@ -5,67 +5,73 @@ import "google/protobuf/descriptor.proto";
 
 message Foo {
   string name = 1;
-  Foo foo = 2 [
```

Comment:
In v26, a delimited field could only be referenced in the text format (and thus in a message literal) if its name == lower-cased type-name, and then only via the type name (using the lower-cased type name in the message literal would fail).

In v27, a message literal is only allowed to be referenced by the type name if the field "looks like a group" -- meaning a delimited field where _not only_ is the field name == lower-cased type name, _but also_ where the type is a sibling of the field (declared in the same scope).

In v27, any group or delimited field name can also be referenced by the actual field name (the lower-cased name). But in v26, delimited fields/groups cannot and can only be referenced by the type name. (This means that v26 has a serious bug that if a field's name != lower-cased type name, it cannot be referenced in the text format.)

So this re-organizes this file so it can actually be compiled using _either_ version. That is necessary because protocompile complies with the rules in v27. But there isn't a release candidate for v27 yet, so we also need to be able to compile the file with v26 of protoc (in order to compare the outputs of protoc and protocompile, which is how this file gets used in tests).